framework/src/problems/FEProblemBase.C

```diff
@@ -8376,9 +8376,12 @@ FEProblemBase::checkProblemIntegrity()
   checkCoordinateSystems();
 
   // Coordinate transforms are only intended for use with MultiApps at this time. If you are not
-  // using multiapps but still require these, contact a moose developer
-  if (_mesh.coordTransform().hasScalingOrRotationTransformation() && _app.isUltimateMaster() &&
-      !hasMultiApps())
+  // using multiapps but still require these, specify the '_check_length_unit_parameter' in your
+  // application as a private parameter to allow skipping this check
+  if ((_mesh.coordTransform().hasScalingOrRotationTransformation() && _app.isUltimateMaster() &&
+       !hasMultiApps()) &&
+      (!_app.isParamValid("_check_length_unit_parameter") ||
```

Comment:
Setting this parameter will pass this check for all transform parameters without multiapps.

Reply:
Yes it will apply a scaling on any transfer that is not disabling this feature 
Many if not all are disabling it by default though

Reply:
What is your experience with people coupling griffin to other codes? Does it usually cause a problem when griffin is in cm ? Is the power density in W/cm3 then and people get the conversion wrong? Do they map regions wrong in the transfer because the coordinate scaling is off by default?
Or are they loading multigroup cross sections from tools in cm and they get ridiculous mfps?
I like Jan's solution of having the code complain about ridiculous mfps.

Reply:
We manually switch meter when coupling other physics. I am not aware of auto transform being used during the transfer in multiphysics models.

Cross sections are populated in materials `computeQpProperties` currently. We will need a new interface function to evaluate the element mfps right after material construction (which for example, using reference temperature to evaluate xs instead of using temperature variable values on qps). Or we can evaluate mfps after an element loop where materials `computeQpProperties` are called.

Reply:
> We manually switch meter when coupling other physics. I am not aware of auto transform being used during the transfer in multiphysics models.

If you switched the mesh unit to cm i think the approach that everybody got used to will now require an additional parameter to work? That seems problematic. 

> A XS sanity check uo is a nice feature.

I agree

Reply:
Yes, I feel we might need a separate parameter for the length unit switch because `Mesh/lengh_unit` might be used for transformation. Maybe we can add a flag here like `Mesh/set_length_unit_transform_for_transfers`? In this way, we can optionally ignore the length unit transformation for multiapps.

Reply:
>  In this way, we can optionally ignore the length unit transformation for multiapps.

we can already ignore the length unit on a per-transfer basis so this is not necessary.

Reply:
nevermind we cannot ignore the length_unit in transfers at this time. 

I don't perceive a huge desire to change the length unit. I think preferring SI is sensible and appreciated by non-reactor physicists. Did you hear from collaborators that we need to change.

Reply:
When doing multiphysics, we typically let Griffin run with meter by setting `is_meter` parameter for several places currently. When doing so, the unit is in consistent with the default unit meter in `Mesh/length_unit`. It looks like that once we have `Mesh/length_unit=cm` in the input, Griffin does not need to set `is_meter` and all transfers can work properly including transfers for transferring power density, mesh displacement, heat transfer coefficients, etc. If this is the case, we should freely choose `Mesh/length_unit` (of course to be consistent with the mesh). So it should be safe to use `_mesh.coordTransform().isTransformParamSet()` to skip the check whether multiapps exist when `Mesh/length_unit` is set. And Griffin can default the length unit to cm.